lib/galaxy/datatypes/binary.py

```diff
@@ -1017,30 +1017,30 @@ def set_meta(self, dataset, overwrite=True, **kwd):
             f = h5py.File(dataset.file_name)
```

Comment:
```suggestion
            f = h5py.File(dataset.file_name, 'r')
```

Also use a `with` statement?